src/fromager/build_environment.py

```diff
@@ -88,9 +88,17 @@ def _createenv(self) -> None:
             logger.info("reusing build environment in %s", self.path)
             return
 
+        # use our seeder plugin to install `build` command
         logger.debug("creating build environment in %s", self.path)
         external_commands.run(
-            [sys.executable, "-m", "virtualenv", str(self.path)],
+            [
+                sys.executable,
+                "-m",
+                "virtualenv",
+                "--download",
```

Comment:
Do we always want to use `--download`? Downstream we're relying on the system package for virtualenv and trying to only install tools from our build server for secure builds.

Reply:
The `virtualenv` package bundles wheels in its distribution. To get rid of downloads, we would have to include a `build` wheel in the Fromager wheel. Internally, `virtualenv` downloads with `pip download`. We can redirect download to our mirror by setting an env var.

```shell
$ find virtualenv -name '*.whl'
virtualenv/seed/wheels/embed/pip-24.2-py3-none-any.whl
virtualenv/seed/wheels/embed/setuptools-68.0.0-py3-none-any.whl
virtualenv/seed/wheels/embed/pip-24.0-py3-none-any.whl
virtualenv/seed/wheels/embed/wheel-0.42.0-py3-none-any.whl
virtualenv/seed/wheels/embed/wheel-0.44.0-py3-none-any.whl
virtualenv/seed/wheels/embed/setuptools-74.1.2-py3-none-any.whl
```

Reply:
OK, if we can redirect the downloads we should do that downstream and document how upstream.

I'm not sure how this implementation is technically better than just creating the environment and installing `build` into it ourselves like we do with the other tools. Why do it this way with the plugin?

Reply:
Virtualenv uses some tricks to make installation of seed packages faster. I have created #436.

Reply:
After more experimentation, I came to the conclusion that it is easier to use the same bundling approach as CPython's `ensurepip` and `virtualenv`. They store wheel files in git and ship them to their users.

The wheel files for ``build``, ``packaging``, and `pyproject_hooks` are small and rarely change. Each project has 1-3 releases per year.

Reply:
For setuptools and wheel are we going to end up relying on the wheels that are shipped by virtualenv instead of using the ones that we built?

Reply:
We have always relied on setuptools and wheel package from virtualenv for the initial seeding of a virtualenv.

- `virtualenv` and Python <3.12 seeds a venv with pip, setuptools, and wheel
- `virtualenv` and Python >=3.12 seeds a venv with pip
- `venv` (from CPython) and Python <3.12 seeds a venv with pip and setuptools
- `venv` and Python >=3.12 seeds a venv with pip only